### PR TITLE
fixing multiple deposits without finalisation bug

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

File: db/deposit.go [hunk 1]
```diff
@@ -64,60 +61,58 @@ func (db *DB) FinaliseDeposits(pathToDepositSubTree uint64, depositRoot core.Byt
...
-		return deposits, err
+	result := db.Instance.Find(&deposits)
+	if err := result.Error; err != nil {
+		return err
+	}
+
+	// iterate over all pending deposits and add subtreeID
+	// to deposits who doesnt have it
+	for _, deposit := range deposits {
+		if deposit.SubtreeID == 18446744073709551615 {
```
Comment (ChihChengLiang) [inline db/deposit.go:110]: This looks like 2**64-1. Why do we need this check?

**Review by ChihChengLiang — COMMENTED** [review]
The first round of the review

**Review by ChihChengLiang — APPROVED** [review]
LGTM

File: db/state.go [hunk 2]
```diff
@@ -188,16 +189,27 @@ func (db *DB) updateState(newState core.UserState, path string) error {
...
-	expectedHash := core.DefaultHashes[int(params.MaxDepth)-depth]
+
+	// the expected hash of an empty node at the given depth
+	height := int(params.MaxDepth) - depth
+	expectedHash := core.DefaultHashes[height]
+
+	// iterate through all leaves at the depth to find this empty leaf
+	// NOTE: as we get more deposits this part would need optimisations
+	// for example running splitting the queries in different threads
 	for i := 0; i < totalLeaves; i++ {
```
[the remark this answers is not in the document]
Comment (vaibhavchellani) [reply db/state.go:213]: Good point, let me fix this in the coming PRs

**Pull request merged**